src/deviceOauthService.ts

```diff
@@ -48,7 +48,9 @@ async function makeRequest<T extends JsonMap>(options: HttpRequest): Promise<T>
   const rawResponse = await new Transport().httpRequest(options);
   const response = parseJsonMap<T>(rawResponse.body);
   if (response.error) {
-    const err = new SfError('Request Failed.');
+    const errorDescription = response.error_description ? `(${response.error_description as string})` : '';
+    const error = response.error ? `(${response.error as string})` : 'Unknown';
```

Comment:
This is one solution if we can't put good typings on <T>
```suggestion
    const errorDescription = typeof response.error_description === 'string' ? response.error_description : ''
    const error = typeof response.error === 'string' ?  response.error : 'Unknown';
```